### DISCUSSION TITLE
Setting number of decimals in floorplan

### DISCUSSION THREAD

**Discussion by wavoigt** [ExperienceLovelace/ha-floorplan#447] [General]
How to set the number of decimals of values in the floorplan (in dashboard .yaml or .svg file)?

**Comment by OzGav** [comment]
That will depend on where the number is coming from. Either fix it in a HA template or use JavaScript in the floorplan rules

**Comment by perlchamp** [comment]
same problem. i want to use javascript in the floorplan rules. But HOW?
Did only find template-code :-(

By the way:
how to split yaml for different rules ?

home.yaml:
`rules: !include rules.yaml`

rules.yaml:
`- entity: sensor.esp_8266_ip50_temperatur_wz
    element: sensor.esp_8266_ip50_temperatur_wz
    state_action:
      - service: floorplan.text_set
        service_data: '${entity.state ? entity.state + " °C" : "--"}'

- entity: sensor.esp_8266_ip70_temperatur_ku
    element: sensor.esp_8266_ip70_temperatur_ku
    state_action:
      - service: floorplan.text_set
        service_data: '${entity.state ? entity.state + " °C" : "--"}'`

the second entity i would like to put in another .yaml. Is this possible? If yes, HOW (please with code and where to put)?

I'm wondering when the code will finally be better represented here in Github using the < > tags. This is crap.

Many thanks in advance!

So long
Pc

**Reply by OzGav** [reply]
Here is one example

```
              state_action:
                service: floorplan.text_set
                service_data: '${entity.state != "unavailable" ? (Math.ceil(entity.state * 10) / 10).toFixed(1) + "°C" : "?"}'
```

**Reply by perlchamp** [reply]
hi, thanks.
.toFixed isn't really good method. For me the following works very well:
`'${entity.state ? (Math.round(entity.state * 100)/100) + " W" : "--"}'`
